### PR TITLE
add process title to npm cli

### PR DESCRIPTION
**Pull request by unphased**
See 
https://npm.community/t/why-does-the-cli-override-process-title/8400
https://github.com/tmux-plugins/tmux-resurrect/issues/201

Code from http://broken-by.me/node-js-change-visible-process-name

Not a huge number of tmux resurrect users out there, but there are many reasons for setting the process title. For example, it would make it easier to identify an npm process inside `top`, `htop`, `ps`. We would simply like to give npm the ability to set the title. 

This just adds simple code to report the npm command being run in the process title. 

I am expecting this to be closed because I suspect that the reason why the shell command that is being run is not shown in the title is done for security reasons (in case an npm command is constructed with passwords right in the command). 

I think that even if this is grounds for rejection, we could at least have a discussion about whether we could add a flag to the CLI to allow the title to be manually specified or maybe a flag that can be configured to let it run this code?

Basically this will allow npm cli to report itself as `npm start` or `npm run lint` etc.  rather than just always `npm`.

Thanks

### CODE REVIEW

**Comment by isaacs** [comment]
Yes, the security concerns you bring up are exactly my objection, so good job predicting that ;)

But I think we can and should do something like this.  We could maybe do something more useful by setting `process.title` to something like: `process.title = ['npm', npm.command, ...npm.argv]`, so we just show positional args instead of _all_ arguments to the process.

Also, should be done in v7, not on v6, since we're so close to v7 GA.

**Comment by isaacs** [comment]
Will be addressed in npm v7 by https://github.com/npm/cli/pull/2154
